subprojects/core/src/integTest/groovy/org/gradle/api/tasks/package-info.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@NonNullApi
+package org.gradle.api.tasks;
+
+import org.gradle.api.NonNullApi;
```

Comment:
This changes a lot of things, and will need a thorough review. I think it would be best to do it in a separate PR.

Reply:
I removed both `package-info.java` files.